### PR TITLE
WIP - Upgrade to Net standard

### PR DESCRIPTION
**Pull request by tforkmann**

### CODE REVIEW

File: FSharp.Azure.StorageTypeProvider.sln [hunk 1]
```diff
@@ -1,55 +0,0 @@
-Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
```
Comment (forki) [inline FSharp.Azure.StorageTypeProvider.sln]: Why did you remove the sln files?

File: paket.dependencies [hunk 2]
```diff
@@ -2,7 +2,7 @@ redirects: force
 source https://www.nuget.org/api/v2/
 framework: netstandard2.0
 
-nuget FAKE
```
Comment (forki) [inline paket.dependencies]: Please don't depend on prerelease fake

File: tests/IntegrationTests/BlobUnitTests.fs [hunk 3]
```diff
@@ -74,11 +73,17 @@ let blobMainTests =
             value |> shouldEqual "thing")
 
         testCase "Cloud Blob Client relates to the same data as the type provider" (fun _ ->
-            Expect.contains (Local.Containers.CloudBlobClient.ListContainers() |> Seq.map(fun c -> c.Name)) "samples" "")
+            Expect.contains (
+                Local.Containers.CloudBlobClient.ListContainersSegmentedAsync(null).Result
```
Comment (forki) [inline tests/IntegrationTests/BlobUnitTests.fs]: Instead of using .Result - please use task {}

Comment (tforkmann) [reply tests/IntegrationTests/BlobUnitTests.fs]: We have to do that for the tp itself as well

File: FSharp.Azure.StorageTypeProvider.sln (hunk 1 above)
Comment (tforkmann) [reply FSharp.Azure.StorageTypeProvider.sln]: Sorry that was a mistake ... we be added again

**Comment by tforkmann** [comment]
The workaround was pretty simple: you have to reference the FSharp.Compiler.Tool for the typeprovider as well as to the user of the typeprovider in this case it is the test project

File: tests/IntegrationTests/BlobUnitTests.fs (hunk 3 above)
Comment (isaacabraham) [reply tests/IntegrationTests/BlobUnitTests.fs]: The net standard branch is hacked together at the moment - there are loads of .`Result` scattered around currently that aren't in master. This is because there are breaking changes between the .NET Standard and .NET Framework version of the Azure SDK at the *same* version - literally switching between the two breaks your code.

Just to get things working on .NET Core, I'm happy for .Results to be there at the minute, but before result we'll clean them all up.

**Comment by isaacabraham** [comment]
@tforkmann firstly thank you for working on this - I really appreciate it. Is this Compiler.Tool reference required for all TPs e.g. the SqlProvider etc.? And is there a plan to make it unnecessary?

I'll review this in the coming days - I'm really excited about porting this to Core.

File: .paket/Paket.Restore.targets [hunk 4]
```diff
@@ -38,18 +44,22 @@
...
+    <MSBuild Projects="$(PaketToolsPath)paket.bootstrapper.proj" Targets="Restore" />
+  </Target>
+
+  <Target Name="PaketRestore" Condition="'$(PaketRestoreDisabled)' != 'True'" BeforeTargets="_GenerateDotnetCliToolReferenceSpecs;_GenerateProjectRestoreGraphPerFramework;_GenerateRestoreGraphWalkPerFramework;CollectPackageReferences" DependsOnTargets="PaketBootstrapping">
 
     <!-- Step 1 Check if lockfile is properly restored -->
     <PropertyGroup>
       <PaketRestoreRequired>true</PaketRestoreRequired>
-      <NoWarn>$(NoWarn);NU1603</NoWarn>
+      <NoWarn>$(NoWarn);NU1603;NU1604;NU1605;NU1608</NoWarn>
```
Comment (isaacabraham) [inline .paket/Paket.Restore.targets]: What are these extra NoWarns added for?

File: build.fsx [hunk 5]
```diff
@@ -3,15 +3,20 @@
 // --------------------------------------------------------------------------------------
 
 #r @"packages/FAKE/tools/FakeLib.dll"
-
-open Fake
-open Fake.AppVeyor
+open Fake.BuildServer
```
Comment (isaacabraham) [inline build.fsx]: I'm not against this, but was it necessary to upgrade to FAKE 5 as part of this PR? Are there any breaking changes?

File: global.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (isaacabraham) [inline global.json]: Please remove the `global.json` file - I don't think it's helpful having it in the project.

File: .paket/Paket.Restore.targets (hunk 4 above)
Comment (forki) [reply .paket/Paket.Restore.targets]: This is normal change in paket files. We take care of these ourselves

File: global.json (hunk 6 above)
Comment (forki) [reply global.json]: O_o it's so important.

File: build.fsx (hunk 5 above)
Comment (tforkmann) [reply build.fsx]: I will split the Pull requests.

**Comment by tforkmann** [comment]
The SqlProvider and the other TP using a hack by referencing the fsc.exe for getting the tps working on compiletime. https://github.com/fsprojects/SQLProvider/blob/master/tests/SqlProvider.Core.Tests/MsSql/SqlProvider.Core.Tests.fsproj#L15
I just reference the FSharp.Compiler.Tools instead. That's basically doing the same thing.

File: global.json (hunk 6 above)
Comment (isaacabraham) [reply global.json]: Not in this project it ain't.

**Comment by tforkmann** [comment]
got rid of the global.json now

**Comment by forki** [comment]
I think Isaac's idea is to get rid of the whole dotnet sdk install stuff in
the fake script.

Am Di., 6. Nov. 2018, 11:31 hat Tim Forkmann <notifications@github.com>
geschrieben:

**Comment by tforkmann** [comment]
fine^^

**Comment by tforkmann** [comment]
Should i rewrite all the test to testTasks if possible?

**Comment by forki** [comment]
I think it would be good

Am Di., 6. Nov. 2018, 12:01 hat Tim Forkmann <notifications@github.com>
geschrieben:

**Comment by isaacabraham** [comment]
I've been working to remove the `.Result` everywhere - unfortunately this has highlights several other problems.

One issue is that the Azure SDK is trying to force asynchronous coding on us even though it doesn't fit with type providers - so at some point there needs to be a `.Result` at the top, or a `|> Async.AwaitTask |> Async.RunSynchronously` etc.

The second issue is that scripts are currently broken. Firstly, I think this is caused by not having the "workaround" available in FSI, but not sure - @dsyme are there any netstandard TPs around that work in FSI?

Second issue is that "forcing" the Result of Task in FSI has a nasty habit of locking up FSI completely - from what I remember from a discussion long ago, this is "expected behaviour" of FSI (@dsyme comments?) but it's again causing issues here because at some point we need to jump out of `task { }` to the TP thread. I've tried "switching" to `async { }` using `|> Async.AwaitTask |> Async.RunSynchronously` but not had much joy yet.

**Comment by tforkmann** [comment]
I did some research.

Don create an template for the "new" kind of TypeProviders.
https://github.com/fsprojects/FSharp.TypeProviders.SDK#the-f-type-provider-sdk
The TypeProvider now have to be splitted into DesignTime and RunTime.

Here an example:
https://github.com/fsprojects/OpenAPITypeProvider/tree/master/src

I gonna try that out.

**Comment by isaacabraham** [comment]
@tforkmann good news - I've made some progress with locking issue - it's caused by eagerly evaluating Tasks, but Asyncs don't cause the issue. I've also worked on the blobs and tables side so that they correctly work with the new netstandard version of the APIs. It's not quite there yet, but I have a basic version now up and running.

**Comment by isaacabraham** [comment]
@tforkmann this PR shouldn't have been closed - I just merged it into the main repo because I've been doing some work on it as well. I have no idea how to reopen it - perhaps when you next push code in it'll magically open up again. @forki any ideas?

**Comment by isaacabraham** [comment]
@tforkmann also - please remove the ionide/symbolcache.db file and don't re-commit it - it doesn't belong in the repo.
